nebula-logger/main/log-management/lwc/loggerSettings/__tests__/loggerSettings.test.js

```diff
@@ -0,0 +1,551 @@
+    await Promise.resolve();
+
+    return loggerSettingsElement;
+}
+
+test.todo('test for search term too short (2 characters or less)');
+test.todo('test for no search results');
+test.todo('test for handleRecordSearchBlur()');
+test.todo('test for handleRemoveSetupOwner()');
+test.todo('test for invalid inputs when saving');
+test.todo('test for error handling via toast messages');
+
+describe('Logger Settings lwc tests', () => {
+    afterEach(() => {
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+        jest.clearAllMocks();
+    });
+
+    it('loads component without modify access', async () => {
+        const loggerSettingsElement = await initializeElement(false);
+
+        // Check the component
+        expect(document.title).toEqual('Logger Settings');
+        // expect(getObjectInfo.getLastConfig()).toEqual(mockObjectInfo);
+        const refreshBtn = loggerSettingsElement.shadowRoot.querySelector('lightning-button[data-id="refresh-btn"]');
+        expect(refreshBtn).toBeTruthy();
+        const newRecordBtn = loggerSettingsElement.shadowRoot.querySelector('lightning-button[data-id="new-btn"]');
+        expect(newRecordBtn).toBeFalsy();
+        const datatable = loggerSettingsElement.shadowRoot.querySelector('lightning-datatable');
+        expect(datatable).toBeTruthy();
+        expect(datatable.data).toEqual(mockRecords);
+    });
+
+    it('loads component with modify access', async () => {
+        const loggerSettingsElement = await initializeElement(true);
+
+        // Verify the expected Apex/framework calls
+        expect(canUserModifyLoggerSettings).toHaveBeenCalledTimes(1);
+        expect(getPicklistOptions).toHaveBeenCalledTimes(1);
+        expect(getRecords).toHaveBeenCalledTimes(1);
+        expect(createRecord).toHaveBeenCalledTimes(0);
+
+        // Check the component
+        expect(document.title).toEqual('Logger Settings');
+        const refreshBtn = loggerSettingsElement.shadowRoot.querySelector('lightning-button[data-id="refresh-btn"]');
+        expect(refreshBtn).toBeTruthy();
+        const newRecordBtn = loggerSettingsElement.shadowRoot.querySelector('lightning-button[data-id="new-btn"]');
+        expect(newRecordBtn).toBeTruthy();
+        const datatable = loggerSettingsElement.shadowRoot.querySelector('lightning-datatable');
+        expect(datatable).toBeTruthy();
+        expect(datatable.data).toEqual(mockRecords);
+    });
+
+    it("shows record modal when 'new' button is clicked", async () => {
+        const loggerSettingsElement = await initializeElement(true);
+        createRecord.mockResolvedValue(mockNewRecord);
+
+        // Clickety click
+        const newRecordBtn = loggerSettingsElement.shadowRoot.querySelector('lightning-button[data-id="new-btn"]');
+        newRecordBtn.click();
+        // Yes, 2 calls for Promise.resolve() are needed... because of... reasons?
```

Comment:
I really love this comment

Reply:
It turned out to be a necessary comment 😆 - since I was working on this PR for a while, I kept forgetting that I _do_ need both, so I kept removing one of the lines on different days